### ISSUE TITLE
[Bug] Changing Special Shops (Alpha Sapphire) *sometimes* makes the ROM unusable.

### ISSUE THREAD

**Issue by Rukir** [Ajarmar/universal-pokemon-randomizer-zx#248]
I am using a US version of Alpha Sapphire. Me and my friend are trying to do a nuzlocke race, so I send them a copy of my randomizer settings, which notably includes randomizing shops. Somewhat oddly, theirs works, and mine doesn't. In fact, no matter how many times I re-apply and re-download a new randomization, it always closes whenever I open it in Citra (latest nightly version, but I tried other versions as well and the problem persisted.)

After much trial and error, I found that this problem lies SOLELY on enabling "Ban Bad Items" in "Special Shops". No other option, whether set individually or combined with others, caused this problem.

Like I said earlier, oddly, my friend's rom worked first try; it was my exact settings, and when they sent me the ROM for me to download and use, lo and behold, it worked. This has me totally stumped; I tried randomizing Omega Ruby instead, different versions of Citra, and different versions of the randomizer, but this problem was consistent until I disabled Ban Bad Items in that category. I have no idea what to send for this, so if you need any files just ask.

**System (please complete the following information):**
 - Operating System: Windows 10.
 - Java Version: Version 8, Update 291
 - Randomizer Version: 4.2.1

UPDATE: It actually happens when I mess with the shops at all.

**Comment by tom-overton** [comment]
Can you please supply a log for when you randomize your ROM? Without a log, there's not really much I can do.

**Comment by lilleypad13** [comment]
Generated Log File:
[PokemonAlphaSaph_FirstRando_ZX.cxi.log](https://github.com/Ajarmar/universal-pokemon-randomizer-zx/files/7041293/PokemonAlphaSaph_FirstRando_ZX.cxi.log)

Image of Where It Gets Stuck:
![ASRandomizerIssue](https://user-images.githubusercontent.com/35742090/130670542-fc133246-72ac-4136-bd5a-dd6fdc100cc7.png)

System:
Operating System: Windows 10
Java Version: 1.8.0_301 (Version 8 Update 301)
Randomizer Version: 4.2.1

I actually ran into a similar issue I believe. I was testing this randomizer on the newer generations so I was modifying a lot of settings just to see how they worked out, and it was going well when I checked: Pokemon X and Pokemon Ultra Sun. However, the Pokemon Alpha Sapphire rom appeared to be created and randomized, but would get stuck on launching when trying to open the .cxi file with Citra (Citra Nightly 1721).

Upon seeing this thread I tried using all the same settings but removing the changes to the shop, and as observed by Rukir, this also seemed to fix my issue and it ran no problem and properly randomized as far as I saw. I have only tried randomizing Alpha Sapphire these two times so far so testing has been limited so far, but it seems to matchup with what was seen here.

If there is any more information I can provide or that log file isn't what you're looking for, feel free to ask, I'm happy to help!
